docs/stack/insight/tutorials/insight-stream-consumer.md

```diff
@@ -0,0 +1,199 @@
+The point of a stream is that it's not going to end, so you cannot capture whole datasets and do some processing on them. 
+In order to process the data, a stream is separated into different roles. 
+A _consumer_ reads from the stream and does some work on it. 
+For example, if the temperature is above a certain threshold, it puts a message out to turn on the air conditioner in that unit or notify the maintenance.
+
+<img src="../images/stream.png">
+
+
+It is possible to have multiple consumers doing different jobs, one measuring humidity, and another taking temperature measurements over periods of time. 
+Redis runs everything in memory that is finite. 
+To avoid runaway data, streams can be trimmed when you add something to them. 
+You can specify to remove everything but a number of last entries or remove stream that has an ID lower than specified, for example a timestamp. 
+You can also expire streams using keys by writing to a different key each day and expiring those keys after a period of time, say a week. 
+An ID can be any number.
+
+## Adding new entries 
+
+Use `XADD` with `*` for ID and that automatically generates timestamps plus fields and values. There are a couple of ways of retrieving things. You could give it a time range or you could ask for everything that's happened since a timestamp or ID that you specify. On one command you can ask for anything from 10:30 until 11:15 am on each day over a specified period of time.
+
+## Consumer groups
+
+A more realistic use case would be a system with many temperature sensors whose data Redis puts in a stream, records the time they arrive, and orders them. 
+
+<img src="../images/consumer.png">
+
+On the right side we have two consumers that read the stream. One of them is alerting if the temperature is over a certain number and texting the maintenance crew that they need to do something, and the other is a data warehouse that is taking the data and putting it into a database. 
+
+They run independently of each other. 
+Up in the right, we have another sort of task. 
+Let's assume that alerting and data warehouse are really fast. 
+You get a message whether the temperature is larger than a specific value, which might take a millisecond. 
+And alerting can keep up with the data flow. 
+One way you can scale consumers is _consumer groups_, which allows multiple instances of the same consumer or same code to work as a team to process the stream.
+
+## Managing streams in RedisInsight
+
+You can add a stream in RedisInsight in two ways: create a new stream or run an existing stream.
+
+To create a stream, start by selecting the key type (stream). 
+You cannot set time to live (TTL) because it cannot be put on a message in a stream; it can only be done on a Redis key. Name the stream _mystream_. 
+Then, set the *Entry ID* to `*` to default to timestamp. 
+If this were a supermarket queue, for example, you would set it to a different number. 
```

Comment:
Not sure that this fits here, could be omitted?  Or changed to "If you have your own ID generation strategy, enter the next ID from your sequence.  Remember that the ID must be higher than the ID of any other entry in the stream."